tests/codegen/thread-local.rs

```diff
@@ -4,6 +4,7 @@
 // ignore-wasm globals are used instead of thread locals
 // ignore-emscripten globals are used instead of thread locals
 // ignore-android does not use #[thread_local]
+// ignore-nto does not use #[thread_local]
```

Comment:
With this many ignores, we clearly are heading into "this should be a passlist, not a blocklist" territory, but that's not for this PR.

Reply:
I think having it as a blocklist is still useful, as reviewers will know that it's not supported on that platform.

Reply:
That's a pretty good point, yeah.